src/main/kotlin/com/autonomousapps/Flags.kt

```diff
@@ -45,6 +48,12 @@ object Flags {
       .getOrElse(default)
   }
 
+  internal fun Project.compatibility(): Compatibility {
+    return getGradlePropForConfiguration(FLAG_DISABLE_COMPATIBILITY, Compatibility.WARN.name).let {
+      Compatibility.valueOf(it.toUpperCase(Locale.US))
```

Comment:
Are we ok just throwing an `IllegalArgumentException` if the user passes a bad value or do you think this warrants a nicer error message?

Reply:
```suggestion
      val value = it.toUpperCase(Locale.US)
      Compatibility.values().find { it.name == value } ?: error("Unrecognized value '$it' for 'dependency.analysis.compatibility' property. Allowed values are ${Compatibility.values()}")
```